gradle/plugins/build-logic/src/main/kotlin/rsocket.multiplatform.gradle.kts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2015-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.jetbrains.kotlin.gradle.*
+
+plugins {
+    kotlin("multiplatform")
+    id("build-parameters")
+}
+
+kotlin {
+    jvmToolchain(8)
+
+    @OptIn(ExperimentalKotlinGradlePluginApi::class)
+    targetHierarchy.default()
+
+    targets.configureEach {
+        compilations.configureEach {
+            compilerOptions.configure {
+                freeCompilerArgs.add("-Xrender-internal-diagnostic-names")
+            }
+        }
+    }
+
+    sourceSets.configureEach {
+        languageSettings {
+            progressiveMode = true
```

Comment:
What is progressive mode?

Reply:
https://kotlinlang.org/docs/compiler-reference.html#progressive
Mostly to see compiler errors earlier, when they are introduced as warnings